### PR TITLE
perf: judge whether use domain_matching_func

### PR DESCRIPTION
**Pull request by abichinger**
Improve rbac with domain model performance

### CODE REVIEW

**Comment by hsluoyz** [comment]
@sheny1xuan plz review

File: casbin/rbac/default_role_manager/role_manager.py [hunk 1]
```diff
@@ -289,13 +292,20 @@ def _get_role_manager(self, *domain):
...
+                for domain_str in self.rm_map.keys()
+                if match_error_handler(
+                    self.domain_matching_func, domain_str, domain_pattern
+                )
+            ]
+        else:
+            if domain_pattern in self.rm_map:
+                return [self.rm_map[domain_pattern]]
+            else:
+                return []
```
Comment (sheny1xuan) [inline casbin/rbac/default_role_manager/role_manager.py]: Maybe can return like before `if expression`. `return [self.rm_map[domain_pattern] if domain_pattern in self.rm_map]`.

**Comment by sheny1xuan** [comment]
LGTM!

File: casbin/rbac/default_role_manager/role_manager.py (hunk 1 above)
Comment (abichinger) [reply casbin/rbac/default_role_manager/role_manager.py]: done

**Comment by hsluoyz** [comment]
@sheny1xuan this is your repo. If you think the PR looks good, please merge it :)

**Comment by sheny1xuan** [comment]
Hi, I merged your pr and can't pass `test ci` in `pycasbin`, I have tried to find out the problem, but I failed. Because I can't find the `partial(self.get_role_manager().domain_matching_func, domain)` in the [error information](https://github.com/casbin/pycasbin/runs/4500907365?check_suite_focus=true). Can you help me to fix this error? maybe you can make another `pr` in [pycasbin  repo](https://github.com/casbin/pycasbin).

**Comment by abichinger** [comment]
@sheny1xuan  I couldn't find `partial(self.get_role_manager().domain_matching_func, domain)` either. I'm going to create a PR on the repository of pycasbin and see if it happens again.

**Edit**: Found it! It is from a recent [PR](https://github.com/casbin/pycasbin/pull/226/files).

**Pull request merged**